### PR TITLE
[Bench][Mamba2] Modernise SSD benchmark suite: parametrize, mamba baseline, model-scale params

### PR DESCRIPTION
**Pull request by stelladuyx**
## Summary

- Replace `@Fixture` decorator with `@pytest.mark.parametrize` + explicit typed `-> None` signatures across all 5 files, matching style of `bench_gqa_decode_paged.py`
- Add `mamba_ssm.ops.triton` baseline comparison to `bench_ssd_chunk_scan_fwd`, `bench_ssd_chunk_state_fwd`, `bench_ssd_state_passing_fwd` (falls back to `torch-ref` when mamba-ssm is not installed); `da_cumsum` and `ssd_decode` have no mamba-ssm equivalent
- Add `_to_mamba_inputs()` layout converter for chunk_scan (TileOPs chunked → mamba flat-sequence layout)
- Add Mamba2 model-scale params to `bench_ssd_chunk_scan_fwd` and `bench_ssd_decode`: 130M/370M/780M/1.3B/2.7B with latency/serving/throughput/long-context cases
- Fix `BenchmarkReport.record()`: replace string literal first arg with `op`; normalise `tag=` to `"torch-ref"` / `"mamba"`

## Test plan
- [x] `--collect-only` → 55 tests collected, no errors
- [x] All pre-commit hooks pass
- [x] With mamba-ssm: `tag="mamba"` results appear; without: falls back to `tag="torch-ref"`

### CODE REVIEW

File: benchmarks/ops/bench_da_cumsum_fwd.py [hunk 1]
```diff
@@ -4,6 +4,10 @@
 import torch
 
 from benchmarks.benchmark import BenchmarkBase, BenchmarkReport
+from tests.ops.test_da_cumsum_fwd import (
+    DaCumsumFwdTest,
+    da_cumsum_fwd_ref,
+)
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_da_cumsum_fwd.py]: **Blocking — dead `tests.*` import.**

This entire import block is dead code:
- `DaCumsumFwdTest` is shadowed by the `workloads` import on line 12
- `da_cumsum_fwd_ref` is shadowed by the local function definition on line 15

No name from `tests.*` is actually used. This re-introduces the coupling that #787 eliminated. Delete these 4 lines.

File: benchmarks/ops/bench_da_cumsum_fwd.py [hunk 2]
```diff
@@ -4,6 +4,10 @@
...
 from tileops.ops.da_cumsum_fwd import DaCumsumFwdOp
 from workloads.ops.da_cumsum_fwd import DaCumsumFwdFixture, DaCumsumFwdTest
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_da_cumsum_fwd.py]: **Nit — unused import.** `DaCumsumFwdFixture` is no longer used after replacing `@DaCumsumFwdFixture` with `@pytest.mark.parametrize`. Remove it:

```suggestion
from workloads.ops.da_cumsum_fwd import DaCumsumFwdTest
```

File: benchmarks/ops/bench_ssd_decode.py [hunk 3]
```diff
@@ -4,6 +4,10 @@
 import torch
 
 from benchmarks.benchmark import BenchmarkBase, BenchmarkReport
+from tests.ops.test_ssd_decode import (
+    SsdDecodeTest,
+    ssd_decode_ref,
+)
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_ssd_decode.py]: **Blocking — dead `tests.*` import.**

Same issue as `bench_da_cumsum_fwd.py`:
- `SsdDecodeTest` is shadowed by the `workloads` import on line 12
- `ssd_decode_ref` is shadowed by the local function definition on line 15

Delete these 4 lines.

File: benchmarks/ops/bench_ssd_decode.py [hunk 4]
```diff
@@ -4,6 +4,10 @@
...
 from tileops.ops.ssd_decode import SsdDecodeOp
 from workloads.ops.ssd_decode import SsdDecodeFixture, SsdDecodeTest
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_ssd_decode.py]: **Nit — unused import.** `SsdDecodeFixture` is no longer used. Remove it:

```suggestion
from workloads.ops.ssd_decode import SsdDecodeTest
```

File: benchmarks/ops/bench_ssd_chunk_state_fwd.py [hunk 5]
```diff
@@ -96,10 +114,25 @@ def test_ssd_chunk_state_fwd_bench(
...
-    def baseline(x, Bmat, dt, dA_cumsum, seq_idx):
-        return ssd_chunk_state_fwd_ref(x, Bmat, dt, dA_cumsum, n_groups=n_groups, seq_idx=seq_idx)
-    result_bl = bm.profile(baseline, *inputs)
-    BenchmarkReport.record("ssd_chunk_state_fwd", locals(), result_bl, tag="baseline")
+    if _chunk_state_fwd is not None:
+        x, Bmat, dt, dA_cumsum, seq_idx = inputs
+
+        def mamba_fwd():
+            return _chunk_state_fwd(
+                Bmat.contiguous(),
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_ssd_chunk_state_fwd.py]: **High — likely wrong tensor layout for mamba_ssm baseline.**

The local `ssd_chunk_state_fwd_ref` in this same file explicitly permutes `dt` and `dA_cumsum` from TileOPs `(b, h, c, L)` to computation `(b, c, h, L)` layout:
```python
dA = dA_cumsum.float().permute(0, 2, 1, 3)   # line 33
dt_chunked = dt.float().permute(0, 2, 1, 3)  # line 37
```

But here, `dt` and `dA_cumsum` are passed to `_chunk_state_fwd` with only `.contiguous()` — no permutation. If mamba_ssm expects the same `(b, c, h, L)` layout (which is the standard convention for `nchunks`-first), this baseline produces **silently wrong benchmark numbers**.

The chunk_scan benchmark has a proper `_to_mamba_inputs` converter with documented layout transforms. This file needs the same treatment.

```suggestion
                dt.permute(0, 2, 1, 3).contiguous(),
```

File: benchmarks/ops/bench_ssd_chunk_state_fwd.py [hunk 6]
```diff
@@ -96,10 +114,25 @@ def test_ssd_chunk_state_fwd_bench(
...
+                x.contiguous(),
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_ssd_chunk_state_fwd.py]: Same layout concern — `dA_cumsum` is `(b, h, c, L)` in TileOPs, likely needs `(b, c, h, L)` for mamba_ssm.

```suggestion
                dA_cumsum.permute(0, 2, 1, 3).contiguous(),
```

File: benchmarks/ops/bench_ssd_state_passing_fwd.py [hunk 7]
```diff
@@ -60,10 +78,23 @@ def test_ssd_state_passing_fwd_bench(batch, num_chunks, n_heads, d_state, dtype,
...
-        return ssd_state_passing_fwd_ref(states, dA_chunk_cumsum, initial_states)
-    result_bl = bm.profile(baseline, *inputs)
-    BenchmarkReport.record("ssd_state_passing_fwd", locals(), result_bl, tag="baseline")
+    if _state_passing_fwd is not None:
+        states, dA_chunk_cumsum, initial_states = inputs
+
+        def mamba_fwd():
+            return _state_passing_fwd(
+                states.contiguous(),
+                dA_chunk_cumsum.contiguous(),
```
Comment (Ibuki-wind) [inline benchmarks/ops/bench_ssd_state_passing_fwd.py]: **High — likely wrong tensor layout for mamba_ssm baseline.**

TileOPs `dA_chunk_cumsum` is `(b, h, c)` (confirmed by the local ref function which iterates `dA_chunk_cumsum[:, :, ci]` over chunk index in dim 2). mamba_ssm's `_state_passing_fwd` likely expects `(b, nchunks, nheads)` = `(b, c, h)`, matching the standard mamba convention where chunks come before heads.

```suggestion
                dA_chunk_cumsum.permute(0, 2, 1).contiguous(),
```

**Review by Ibuki-wind — CHANGES_REQUESTED** [review]
## Summary

Model-scale params, mamba_ssm baseline integration, and the parametrize migration are all good ideas. But there are correctness and hygiene issues that need fixing before merge.

### Blocking

1. **Dead `tests.*` imports in two benchmark files** — `bench_da_cumsum_fwd.py` and `bench_ssd_decode.py` both add `from tests.ops.*` imports where every imported name is immediately shadowed (by local function definitions or the `workloads.*` import on the next line). These imports are 100% dead code, but they re-introduce the `tests.*` coupling that #787 eliminated. Remove them entirely.

2. **Missing tensor layout permutations in mamba_ssm baselines** — `bench_ssd_chunk_state_fwd.py` and `bench_ssd_state_passing_fwd.py` pass `dt`/`dA_cumsum`/`dA_chunk_cumsum` to mamba_ssm with only `.contiguous()`, no layout permutation. But the local reference functions in the same files explicitly `permute(0, 2, 1, 3)` these tensors from TileOPs `(b,h,c,L)` to computation `(b,c,h,L)` layout. If mamba_ssm expects the same computation layout, the baselines produce **silently wrong benchmark numbers**. The chunk_scan file has a proper `_to_mamba_inputs` converter — the other two files need equivalent treatment.

### Non-blocking

3. **Unused fixture imports** — `DaCumsumFwdFixture` and `SsdDecodeFixture` are still imported but no longer used after the `@Fixture` → `@pytest.mark.parametrize` migration.

### Action items

1. Delete the `from tests.ops.*` import blocks in `bench_da_cumsum_fwd.py` and `bench_ssd_decode.py`.
2. Clean up `DaCumsumFwdFixture` and `SsdDecodeFixture` from their respective workloads imports.
3. Add layout permutations (or a shared `_to_mamba_inputs`-style converter) to the mamba baselines in `bench_ssd_chunk_state_fwd.py` and `bench_ssd_state_passing_fwd.py`. Document the expected mamba_ssm layout in a comment, like chunk_scan does.
4. Verify the mamba baselines actually produce correct results by running them against the torch-ref baseline on at least one param set.

File: benchmarks/ops/bench_mamba.py [hunk 8]
```diff
@@ -105,50 +166,127 @@ def calculate_flops(self) -> Optional[float]:
...
+        # Writes (float32): out [B, S, H, P]
+        writes = b * S * h * p * 4
         return float(reads + writes)
 
 
-@SSDChunkScanFwdFixture
-def test_ssd_chunk_scan_fwd_bench(batch, num_chunks, chunk_len, n_heads, d_head, d_state, dtype, tune):
-    test = SSDChunkScanFwdTest(batch, num_chunks, chunk_len, n_heads, d_head, d_state, dtype)
+# ---------------------------------------------------------------------------
+# Benchmark parameters
```
Comment (superAngGao) [inline benchmarks/ops/bench_mamba.py]: Could we scale up the dimensions of the benchmark? The current max parameter size is 1.3B, which is considered a very small model nowadays. The same applies to sequence length (seqlen): short context should be 4096, and long context needs to exceed 256k.

**Comment by stelladuyx** [comment]
Changes already merged in pull request 1015.
